boot.go

```diff
@@ -11,11 +11,15 @@ import (
 	"github.com/deis/builder/pkg/cleaner"
 	"github.com/deis/builder/pkg/conf"
 	"github.com/deis/builder/pkg/gitreceive"
-	"github.com/deis/builder/pkg/gitreceive/storage"
 	"github.com/deis/builder/pkg/healthsrv"
 	"github.com/deis/builder/pkg/sshd"
 	"github.com/deis/builder/pkg/sys"
 	pkglog "github.com/deis/pkg/log"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	_ "github.com/docker/distribution/registry/storage/driver/azure"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+	_ "github.com/docker/distribution/registry/storage/driver/gcs"
+	_ "github.com/docker/distribution/registry/storage/driver/s3-aws"
```

Comment:
@stevvooe I certainly haven't counted you out (I was simply speculating), and definitely understand the maintenance overhead of the separate project.

Reading up, I realize that I wasn't clear that we can make progress with, and will submit PRs upstream to either deis/distribution or the new repo if you do decide to split it out. This storage abstraction is useful to us either way, and a new repo would just be a convenience (precisely because the storage layer is not registry specific).

>These are to support back registration. If you find the documentation to be lacking, we are willing to accept a PR.

Great - I'll need to read through the registry codebase more thoroughly to know exactly what you mean by back registration. Mind pointing me to a good starting place?

>There is no reason to continue to maintain it in this manner, other than convenience, so we're open to suggestions.

Just so I'm clear, do you mean there would be no reason to continue to maintain it in this manner if you do split out the project, or is there something I'm missing?

Anyway, I'm happy to help out where possible (beyond the docs mentioned above).